src/nomad_simulations/properties/spectral_profile.py

```diff
@@ -0,0 +1,601 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+from structlog.stdlib import BoundLogger
+from typing import Optional, List, Dict
+import pint
+
+from nomad import config
+from nomad.metainfo import Quantity, SubSection, Section, Context
+
+from ..utils import get_sibling_section
+from ..physical_property import PhysicalProperty
+from ..variables import Energy2 as Energy
+from ..atoms_state import AtomsState
+from .band_gap import ElectronicBandGap
+
+
+class SpectralProfile(PhysicalProperty):
+    """
+    A base section used to define the spectral profile.
+    """
+
+    value = Quantity(
+        type=np.float64,
+        description="""
+        The value of the intensities of a spectral profile in arbitrary units.
+        """,
+    )
+
+    # TODO implement this in PhysicalProperty (similar to `value.shape`)
+    # value_units = Quantity(
+    #     type=str,
+    #     description="""
+    #     Unit using the pint UnitRegistry() notation for the `value`. Example, if the spectra `is_derived`
+    #     from the imaginary part of the dielectric function, `value` are `'F/m'`.
+    #     """,
+    # )
+
+    def __init__(
+        self, m_def: Section = None, m_context: Context = None, **kwargs
+    ) -> None:
+        super().__init__(m_def, m_context, **kwargs)
+        self.rank = []  # ? Is this here or in the attrs instantiation better?
+
+    def is_valid_spectral_profile(self) -> bool:
+        """
+        Check if the spectral profile is valid, i.e., if all `value` are defined positive.
+
+        Returns:
+            (bool): True if the spectral profile is valid, False otherwise.
+        """
+        if (self.value < 0.0).any():
+            return False
+        return True
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        if self.is_valid_spectral_profile() is False:
+            logger.error(
+                'Invalid negative intensities found: could not validate spectral profile.'
+            )
+            return
+
+
+class DOSProfile(SpectralProfile):
+    """
+    A base section used to define the `value` of the `ElectronicDensityOfState` property. This is useful when containing
+    contributions for `projected_dos` with the correct unit.
+    """
+
+    value = Quantity(
+        type=np.float64,
+        unit='1/joule',
+        description="""
+        The value of the electronic DOS.
+        """,
+    )
+
+    def __init__(
+        self, m_def: Section = None, m_context: Context = None, **kwargs
+    ) -> None:
+        super().__init__(m_def, m_context, **kwargs)
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # TODO add normalization for `projected_dos` to extract `name` (if `m_parent` is `ElectronicDOS`
+        # TODO (cont'd) and `entity_ref` is set)
+
+
+class ElectronicDensityOfStates(DOSProfile):
```

Comment:
> I think it is best to develop an electronic structure post-analysis suite, similar to MatID for symmetries.

Yeah it might be, but we need the whole picture first. IMO, we could publish this repo as it is, and later think of some post-processing analysis tools which could be generalized. As of now, it is true that I am trying to keep as much as possible from the former normalization, but improving the schema.

Reply:
Okay, the standalone module is more an idea for down the line (@JFRudzinski would also like your feedback) and a way to get an easy paper. I'm not suggesting this for now! Still, it feels a bit out of place compared to the simpler, and more to-the-point checks that make up most of the other normalizers.

What is relevant at the moment, though, is _stress testing these algos_. We have numerous cases in our repo, where the band structure or DOS extraction fail, and it's painful to see. This has always been a very tricky issue.
Since this PR focuses on migrating and extending the schema, **I implore you to open a new issue where we tackle this aspect**. In the meantime, I'll think up more test cases.